File: src/main/java/edu/spring/td1/controllers/MainController.java
package edu.spring.td1.controllers;

import edu.spring.td1.models.Element;
import org.springframework.stereotype.Controller;
import org.springframework.ui.ModelMap;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.servlet.view.RedirectView;

import java.util.ArrayList;
import java.util.List;

@Controller
@SessionAttributes("items")
public class MainController {

    @ModelAttribute("items")
    public List<Element> getItems(){
        return new ArrayList<>();
    }

    @RequestMapping(value = "/items", method = {RequestMethod.GET, RequestMethod.POST})
    public String show(ModelMap model,
                       @ModelAttribute("items") List<Element> items){
        model.addAttribute("items", items);
        return "items/show";
    }

    @GetMapping("/items/new")
    public String add(){
        return "items/add";
    }

    @PostMapping("/items/addNew")
    public RedirectView store(@RequestParam String name,
                              @ModelAttribute("items") List<Element> items){
        if (items.stream().noneMatch(el -> el.getName().equals(name))){
            Element el = new Element();
            el.setName(name);
            items.add(el);
            return new RedirectView("/items");
        }
        return new RedirectView("/items/new");
    }

    @GetMapping("/items/inc/{nom}")
    public RedirectView inc(@PathVariable(value = "nom")String name,
                            @ModelAttribute("items") List<Element> items){
        for (Element el : items){
            if (el.getName().equals(name)){
                el.setEvaluation(el.getEvaluation() + 1);
                break;
            }
        }        return new RedirectView("/items");
    }

    @GetMapping("/items/dec/{nom}")
    public RedirectView dec(@PathVariable(value = "nom")String name,
                            @ModelAttribute("items") List<Element> items){
        for (Element el : items){
            if (el.getName().equals(name)){
                el.setEvaluation(el.getEvaluation() - 1);
                break;
            }
        }
        return new RedirectView("/items");
    }

    @GetMapping("/items/delete/{nom}")
    public RedirectView delete(@PathVariable(value = "nom")String name,
                               @ModelAttribute("items") List<Element> items){
        for (Element el : items){
            if (el.getName().equals(name)){
                items.remove(el);
                break;
            }
        }
        return new RedirectView("/items");
    }




}

File: src/main/kotlin/edu/spring/td2/controllers/MainController.kt
package edu.spring.td2.controllers

import edu.spring.td2.repositories.GroupRepository
import edu.spring.td2.repositories.OrganizationRepository
import edu.spring.td2.repositories.UserRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.ui.ModelMap
import org.springframework.web.bind.annotation.GetMapping

@Controller
class MainController {

    @Autowired
    lateinit var groupRepository: GroupRepository
    @Autowired
    lateinit var userRepository: UserRepository
    @Autowired
    lateinit var organizationRepository: OrganizationRepository

    @GetMapping("/", "")
    fun index(model: ModelMap) : String {
        model["nGroups"] = groupRepository.count()
        model["nUsers"] = userRepository.count()
        model["nOrgas"] = organizationRepository.count()
        return "index"
    }
}